### PR TITLE
refactored member operator to only use properties from propagated claims

### PR DESCRIPTION
**Pull request by sbryzak**
This PR is a total refactor of the UserAccountController to use the values stored in PropagatedClaims instead of any other property or annotation values that are derived from the user's SSO token.

Fixes https://issues.redhat.com/browse/SANDBOX-431

E2E tests: https://github.com/codeready-toolchain/toolchain-e2e/pull/858

### CODE REVIEW

File: controllers/useraccount/useraccount_controller.go [hunk 1]
```diff
@@ -237,19 +237,17 @@ func (r *Reconciler) ensureUser(ctx context.Context, config membercfg.Configurat
...
+		expectedIdentities = append(expectedIdentities, commonidentity.NewIdentityNamingStandard(userAcc.Spec.PropagatedClaims.OriginalSub, config.Auth().Idp()).IdentityName())
 	}
 
 	// Also if the sso-user-id annotation is set, then another additional identity is required if it is a different value to the Spec.UserID
-	if val, ok := userAcc.Annotations[toolchainv1alpha1.SSOUserIDAnnotationKey]; ok {
-		if val != userAcc.Spec.UserID {
-			expectedIdentities = append(expectedIdentities, commonidentity.NewIdentityNamingStandard(
-				userAcc.Annotations[toolchainv1alpha1.SSOUserIDAnnotationKey], config.Auth().Idp()).IdentityName())
-		}
+	if userAcc.Spec.PropagatedClaims.UserID != "" && userAcc.Spec.PropagatedClaims.UserID != userAcc.Spec.UserID {
```
Comment (alexeykazakov) [inline controllers/useraccount/useraccount_controller.go:248]: Should be the following instead, right?
```suggestion
	if userAcc.Spec.PropagatedClaims.UserID != "" && userAcc.Spec.PropagatedClaims.UserID != userAcc.Spec.PropagatedClaims.Sub {
```

File: controllers/useraccount/useraccount_controller.go [hunk 2]
```diff
@@ -635,13 +627,11 @@ func (r *Reconciler) updateStatusConditions(ctx context.Context, userAcc *toolch
...
-	if userAcc.Spec.OriginalSub != "" {
-		identities = append(identities, commonidentity.NewIdentityNamingStandard(userAcc.Spec.OriginalSub, config.Auth().Idp()).IdentityName())
+	if userAcc.Spec.PropagatedClaims.OriginalSub != "" {
+		identities = append(identities, commonidentity.NewIdentityNamingStandard(userAcc.Spec.PropagatedClaims.OriginalSub, config.Auth().Idp()).IdentityName())
 	}
-	if val, ok := userAcc.Annotations[toolchainv1alpha1.SSOUserIDAnnotationKey]; ok {
-		if val != userAcc.Spec.UserID {
-			identities = append(identities, commonidentity.NewIdentityNamingStandard(userAcc.Annotations[toolchainv1alpha1.SSOUserIDAnnotationKey], config.Auth().Idp()).IdentityName())
-		}
+	if userAcc.Spec.PropagatedClaims.UserID != "" && userAcc.Spec.PropagatedClaims.UserID != userAcc.Spec.PropagatedClaims.OriginalSub {
```
Comment (alexeykazakov) [inline controllers/useraccount/useraccount_controller.go:633]: If I'm reading it right it should be:
```suggestion
	if userAcc.Spec.PropagatedClaims.UserID != "" && userAcc.Spec.PropagatedClaims.UserID != userAcc.Spec.PropagatedClaims.Sub {
```

File: controllers/useraccount/useraccount_controller.go (hunk 1 above)
Comment (MatousJobanek) [reply controllers/useraccount/useraccount_controller.go:248]: you could update the unit tests so the original fields are empty to make sure that only the new ones are used

File: controllers/useraccount/useraccount_controller.go (hunk 2 above)
Comment (MatousJobanek) [reply controllers/useraccount/useraccount_controller.go:633]: I'm wondering why unit tests didn't fail for that.

File: controllers/useraccount/useraccount_controller.go (hunk 1 above)
Comment (sbryzak) [reply controllers/useraccount/useraccount_controller.go]: Good catch, fixed thanks.

File: controllers/useraccount/useraccount_controller.go (hunk 2 above)
Comment (sbryzak) [reply controllers/useraccount/useraccount_controller.go]: Fixed

**Review by alexeykazakov — APPROVED** [review]

File: controllers/useraccount/useraccount_controller.go (hunk 1 above)
Comment (MatousJobanek) [reply controllers/useraccount/useraccount_controller.go]: @sbryzak but I don't see that the tests would be updated so the original fields would be set to empty to make sure that they are not used at all

**Comment by MatousJobanek** [comment]
@sbryzak you missed some occurrences of the usages of the email annotation?
https://github.com/codeready-toolchain/member-operator/blob/2931fb137d4c73385324160a4538eb0647f142d2/controllers/useraccount/useraccount_controller.go#L397
https://github.com/codeready-toolchain/member-operator/blob/2931fb137d4c73385324160a4538eb0647f142d2/controllers/useraccount/useraccount_controller_test.go#L1554
https://github.com/codeready-toolchain/member-operator/blob/2931fb137d4c73385324160a4538eb0647f142d2/controllers/useraccount/useraccount_controller_test.go#L91

So, as I mentioned in the other comment: https://github.com/codeready-toolchain/member-operator/pull/507#discussion_r1407753800 could you please update the test so it doesn't use the old fields and annotations that we are migrating from?
https://github.com/codeready-toolchain/member-operator/blob/2931fb137d4c73385324160a4538eb0647f142d2/controllers/useraccount/useraccount_controller_test.go#L1609-L1614

**Pull request merged**